ui/app/AppLayouts/Wallet/services/dapps/DAppsRequestHandler.qml

```diff
@@ -324,28 +342,110 @@ QObject {
+                tx = d.getTxObject(method, data)
+            }
+
+            let Math = SQUtils.AmountsArithmetic
+            var gasLimit = Math.fromString("21000")
+            var gasPrice, maxFeePerGas, maxPriorityFeePerGas
+            // Beware, the tx values are standard blockchain hex big number values; the fees values are nim's float64 values, hence the complex conversions
+            if (!!tx.maxFeePerGas && !!tx.maxPriorityFeePerGas) {
+                let maxFeePerGasDec = root.store.hexToDec(tx.maxFeePerGas)
+                gasPrice = Math.fromString(maxFeePerGasDec)
+                // Source fees info from the incoming transaction for when we process it
+                maxFeePerGas = maxFeePerGasDec
+                let maxPriorityFeePerGasDec = root.store.hexToDec(tx.maxPriorityFeePerGas)
+                maxPriorityFeePerGas = maxPriorityFeePerGasDec
+            } else {
+                let fees = root.store.getSuggestedFees(chainId)
+                maxPriorityFeePerGas = fees.maxPriorityFeePerGas
+                if (fees.eip1559Enabled) {
+                    if (!!fees.maxFeePerGasM) {
+                        gasPrice = Math.fromString(fees.maxFeePerGasM)
+                        maxFeePerGas = fees.maxFeePerGasM
+                    } else if(!!tx.maxFeePerGas) {
+                        let maxFeePerGasDec = root.store.hexToDec(tx.maxFeePerGas)
+                        gasPrice = Math.fromString(maxFeePerGasDec)
+                        maxFeePerGas = maxFeePerGasDec
+                    } else {
+                        console.error("Error fetching maxFeePerGas from fees or tx objects")
+                        return
+                    }
+                } else {
+                    if (!!fees.gasPrice) {
+                        gasPrice = Math.fromString(fees.gasPrice)
+                    } else {
+                        console.error("Error fetching suggested fees")
+                        return
+                    }
+                }
+                gasPrice = Math.sum(gasPrice, Math.fromString(fees.l1GasFee))
+            }
+
+            let maxFees = Math.times(gasLimit, gasPrice)
+            return {maxFees, maxFeePerGas, maxPriorityFeePerGas, gasPrice}
+        }
+
+        function getEstimatedFeesStatus(data, method, chainId) {
+            let Math = SQUtils.AmountsArithmetic
+
+            let feesInfo = getEstimatedMaxFees(data, method, chainId)
+
+            let maxFeesEth = Math.div(feesInfo.maxFees, Math.fromString("1000000000"))
+
+            // TODO #15192: extract account.balance
+            //let accountFundsEth = account.balance
+            //let haveEnoughFees = Math.cmp(accountFundsEth, maxFeesEth) >= 0
+            let haveEnoughFees = true
+
+            let maxFeesEthStr = maxFeesEth.toString()
+            let fiatMaxFees = root.currenciesStore.getFiatValue(maxFeesEthStr, Constants.ethToken)
+            let symbol = root.currenciesStore.currentCurrencySymbol
```

Comment:
I think we should use the currency short name. Sorry, probably the Sign modal API is misleading here.